ml/BitBufferCounter.h

```diff
@@ -0,0 +1,54 @@
+// SPDX-License-Identifier: GPL-3.0-or-later
+
+#ifndef BIT_BUFFER_COUNTER_H
+#define BIT_BUFFER_COUNTER_H
+
+#include "ml-private.h"
+
+namespace ml {
+
+class BitBufferCounter {
+public:
+    BitBufferCounter(size_t Capacity) : V(Capacity, 0), NumSetBits(0), N(0) {}
+
+    std::vector<bool> getBuffer() const;
+
+    void insert(bool Bit);
+
+    void print(std::ostream &OS) const;
+
+    bool isFilled() const {
+       return N >= V.size();
+    }
+
+    size_t numSetBits() const {
+        return NumSetBits;
+    }
+
+private:
+    inline size_t size() const {
+        return N < V.size() ? N : V.size();
+    }
+
+    inline size_t start() const {
+        if (N <= V.size())
+            return 0;
```

Comment:
Not sure I understand what you mean? What's the performance overhead here?

Reply:
There is no performance overhead in the current code for this, only that many returns from the middle of functions is not recommended (by misrac standards), as it would complicate the further maintenance.

Reply:
We don't follow this guideline in the codebase. I will change this one if you want though.

Reply:
In general, I welcome it, but the suggestion is let me do it later :)